x-pack/test/api_integration/apis/transform/transforms.ts

```diff
@@ -0,0 +1,137 @@
+      typeOfVersion: 'string',
+      typeOfCreateTime: 'number',
+    },
+    apiTransformTransformsTransformId: {
+      count: 1,
+      transform1: { id: 'the-transform-1', destIndex: 'user-the-transform-1' },
+      typeOfVersion: 'string',
+      typeOfCreateTime: 'number',
+    },
+  };
+
+  async function createTransform(transformId: string) {
+    const config = generateTransformConfig(transformId);
+    await transform.api.createTransform(transformId, config);
+  }
+
+  describe('/api/transform/transforms', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await transform.testResources.setKibanaTimeZoneToUTC();
+      await createTransform('the-transform-1');
+      await createTransform('the-transform-2');
+    });
+
+    after(async () => {
+      await transform.api.cleanTransformIndices();
+    });
+
+    describe('/transforms', function () {
+      it('should return a list of transforms', async () => {
+        const { body } = await supertest
+          .get('/api/transform/transforms')
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send()
+          .expect(200);
+
+        expect(body.count).to.eql(expected.apiTransformTransforms.count);
+        expect(body.transforms).to.have.length(expected.apiTransformTransforms.count);
+
+        const transform1 = body.transforms[0];
+        expect(transform1.id).to.eql(expected.apiTransformTransforms.transform1.id);
+        expect(transform1.dest.index).to.eql(expected.apiTransformTransforms.transform1.destIndex);
+        expect(typeof transform1.version).to.eql(expected.apiTransformTransforms.typeOfVersion);
+        expect(typeof transform1.create_time).to.eql(
+          expected.apiTransformTransforms.typeOfCreateTime
+        );
+
+        const transform2 = body.transforms[1];
+        expect(transform2.id).to.eql(expected.apiTransformTransforms.transform2.id);
+        expect(transform2.dest.index).to.eql(expected.apiTransformTransforms.transform2.destIndex);
+        expect(typeof transform2.version).to.eql(expected.apiTransformTransforms.typeOfVersion);
+        expect(typeof transform2.create_time).to.eql(
+          expected.apiTransformTransforms.typeOfCreateTime
+        );
+      });
+
+      it('should return 200 for transform view-only user', async () => {
+        await supertest
+          .get(`/api/transform/transforms`)
+          .auth(
+            USER.TRANSFORM_VIEWER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_VIEWER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send()
+          .expect(200);
+      });
+    });
+
+    describe('/transforms/{transformId}', function () {
+      it('should return a specific transform configuration', async () => {
```

Comment:
Might also be good to add a test for `get non-existent transform`.

Reply:
Added in c6f388f.